pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -1200,17 +1199,23 @@ func (m *kubeGenericRuntimeManager) computeInitContainerActions(pod *v1.Pod, pod
 				}
 
 				// If the init container is in unknown state, restart it.
-				changes.ContainersToKill[status.ID] = containerToKillInfo{
-					name:      container.Name,
-					container: container,
-					message: fmt.Sprintf("Init container is in %q state, try killing it before restart",
-						status.State),
-					reason: reasonUnknown,
-				}
-				changes.InitContainersToStart = append(changes.InitContainersToStart, i)
+				message = fmt.Sprintf("%s is in %q state, try killing it before restart", message, status.State)
+				reason = reasonUnknown
+				restartContainer = true
 			}
 		}
 
+		if restartContainer {
+			changes.ContainersToKill[status.ID] = containerToKillInfo{
+				name:      container.Name,
+				container: container,
+				message:   message,
+				reason:    reason,
+			}
+			changes.InitContainersToStart = append(changes.InitContainersToStart, i)
+			klog.V(4).InfoS("Message for Init Container of pod", "containerName", container.Name, "containerStatusID", status.ID, "pod", klog.KObj(pod), "containerMessage", message)
```

Comment:
```suggestion
			klog.V(5).InfoS("Message for Init Container of pod", "containerName", container.Name, "containerStatusID", status.ID, "pod", klog.KObj(pod), "containerMessage", message)
```

This is a new log message. I worry we may flood the logs. Maybe try level 5 to be safer